### ISSUE TITLE
Crashing After Rendering 1 to 2 frames

### ISSUE THREAD

**Issue by totti1st** [shteeve3d/blender-wiggle-2#31]
The add-on crashes blender when rendering, after a frame or two, the render has no problem after I disable the add-on

**Comment by shteeve3d** [comment]
this is unfortunately a hard limitation of the addon and the way blender handles data internally. you have to bake and mute the live effect before rendering. i had tried to add logic so it auto mutes itself when rendering, but i've found it still seems to get into a crashable state. i can at least look into that aspect of the issue. another way to usually avoid crashing is to use blender's 'lock interface' option for rendering.

i welcome any input from developers experienced working with python frame handlers during render. the issue best as i can see has to do with renders creating a separate copy of data in memory that is differently evaluated than the viewport, and it is really difficult accessing and writing to the proper versions of data, especially when you get into dependencies of parenting, constraints, colliders, etc. i'd managed to get simple cases working for renders, but it would fall apart with any level of scene complexity.

also worth noting, since the live effect is evaluated based on frame change updates, it doesn't work with motion blur which skips around during render evaluating previous and next frame motion vectors. it would need to maintain a cache of already evaluated frames, which, beyond a point, is just adding a whole bunch of data management complexity that basically is reinventing what baking would do.

**Comment by mike-d3v** [comment]
I think it would be a good idea to add some sort of warning into the add-on which tells the user to bake and disable before rendering, because I spent a very long time figuring out why my renders were crashing before I found out it was caused by the wiggle add-on 🥲

**Comment by emptybraces** [comment]
Same problem encountered.
Crash never happened if disable addon, but it was randomely or miracle idea that I did.
It would be great if it could be fixed in some way, still thanks for the great addon!

**Comment by Paragon0505** [comment]
I had this problem too. Didn't think to bake it. I'll try that.

**Comment by DrummyCrane** [comment]
I had the same problem and my workaround was adding `if scene.wiggle.is_rendering: return` at the beginning of `wiggle_pre`. It worked for my simple project that includes several Wiggle bones so far.
By the way, removing `if scene.wiggle.is_rendering: return` from both `wiggle_pre` and `wiggle_post` allowed me to simulate Wiggle in the final render but I believe it will make problems because on the other [issue](https://github.com/shteeve3d/blender-wiggle-2/issues/18#issuecomment-1585177149), @shteeve3d commented that it was intended to be only available at the viewport.

**Comment by shteeve3d** [comment]
Oh thanks for this! I’ll have to see if there’s any unintended consequences to adding that line to wiggle pre, but maybe it’s just something I overlooked!

**Comment by kissshot55** [comment]
How do I get two chains to self-collide, and the simulation becomes extremely unstable when I put the chains in the collision set

**Comment by leukick** [comment]
Yeah if anyone sees this, renders crash every time after a few frames in Blender 4.0.

I'm using this on a client project so it's very scary and disheartening for it to not render at all near due date. I would buy this as an addon if it can work in Blender 4.0 and beyond though, it is brilliant and extremely useful. 

I tried all of the previous mentioned troubleshooting, the baking + muting the wiggle, disabling the addon, the lock interface in blender before render, and trying a the previous version of the addon (2.2.2 from 2.2.3). I'm surprised it still crashed after completely removing the addon (maybe it still lingers in the codes? IDK how this stuff works).

There's another addon just like this but I refuse to switch because I immediately fell in love with this one, plus I already nearly completed a project with it and can't imagine starting it all over, especially near the due date.

**Comment by DrummyCrane** [comment]
I rendered once in Blender 4.0.0 and 4.0.2 with Wiggle2 2.2.3 for a test but it was ok. But idk what if with other add-ons or in a bigger complicated project.

Wiggle seems to leave some properties such as `bone head` or `bone tail`. But its codes should be removed after a user **remove** the addon in the preference screen, and it would not be run.
